### PR TITLE
feat: add support for output modifier flags

### PR DESCRIPTION
**Pull request by ruicalheno133**
Currently, only `-e` option is used to execute an inline script. 

This PR:
- Adds support for osascript's output modifier flags `-s` (as specified [here](https://ss64.com/osx/osascript.html)). 
- Adds tests to ensure current behaviour is kept and `-s` flag behaviour works

I'm not sure if this solves #4, but using the `-s s` flag, it returns a string that can be parsed into an array. Neverthless, I think it is useful to have this option available to mirror the behaviour of the `osascript` command,.

Output without `-s` flag:
```
$ osascript -e 'return {{"foo", {"bar"}}}'        
foo, bar
```

Output with `-s` flag:
```
$ osascript -e 'return {{"foo", {"bar"}}}'  -s s
{{"foo", {"bar"}}}
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
The point of this package is to abstract away the command-line flags. I think this should instead be an option called `humanReadableReturnValue` that can be set to false to get the wanted behavior.

**Comment by ruicalheno133** [comment]
Got it 👍 Will update accordingly.

As a follow-up, I was wondering if it wouldn't be useful to also be able to read from a file. I could create a new function specifically for that or add a new argument `fromFile` and if this was set to true it would consider the `script` param as a filename. What do you think?

**Comment by sindresorhus** [comment]
> As a follow-up, I was wondering if it wouldn't be useful to also be able to read from a file. I could create a new function specifically for that or add a new argument fromFile and if this was set to true it would consider the script param as a filename. What do you think?

I think the most common use-case for this package are short scripts you can just have inline. If you need to pass a file, you can simply call `fs.readFile`.

**Comment by ruicalheno133** [comment]
Right! Let's keep it simple. 

I added a new argument `humanReadableOutput` to the existing functions in 282e882. 
I thought wrapping it within an options object, but it seemed overkill since there are no more options to be defined.

**Comment by sindresorhus** [comment]
> I thought wrapping it within an options object, but it seemed overkill since there are no more options to be defined.

I think it should be an options-object.

`runApplescript(..., false)` is not very readable.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/typescript-definition-style-guide#documentation

File: index.d.ts [hunk 1]
```diff
@@ -14,14 +24,27 @@ const result = await runAppleScript('return "unicorn"');
...
+import {runAppleScript} from 'run-applescript';
+
+const options = {
+    humanReadableOutput: false
+}
+const result = await runAppleScript('return "unicorn"', options);
+
+console.log(result);
+//=> '"unicorn"'
+```
```
Comment (sindresorhus) [inline index.d.ts]: The example should be on the docs for the `humanReadableOutput` option. Line 8.

File: index.d.ts [hunk 2]
```diff
@@ -14,14 +24,27 @@ const result = await runAppleScript('return "unicorn"');
...
 //=> 'unicorn'
 ```
+
+@example
+```
+import {runAppleScript} from 'run-applescript';
+
+const options = {
+    humanReadableOutput: false
+}
```
Comment (sindresorhus) [inline index.d.ts]: This can just be inline in the function call.

File: index.js [hunk 3]
```diff
@@ -1,20 +1,28 @@
 import process from 'node:process';
 import execa from 'execa';
 
-export async function runAppleScript(script) {
+const DEFAULT_OPTIONS = {
+	humanReadableOutput: true,
+};
+
+export async function runAppleScript(script, options = DEFAULT_OPTIONS) {
```
Comment (sindresorhus) [inline index.js]:
```suggestion
export async function runAppleScript(script, {humanReadableOutput = true} = {}) {
```

File: index.js [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
 	if (process.platform !== 'darwin') {
 		throw new Error('macOS only');
 	}
 
-	const {stdout} = await execa('osascript', ['-e', script]);
+	const outputArg = options.humanReadableOutput ? [] : ['-ss'];
```
Comment (sindresorhus) [inline index.js]:
```suggestion
	const outputArguments = options.humanReadableOutput ? [] : ['-ss'];
```

File: test.js [hunk 5]
```diff
@@ -8,3 +8,19 @@ test('async', async t => {
 test('sync', t => {
 	t.is(runAppleScriptSync('return "unicorn"'), 'unicorn');
 });
+
+test('non human readable output async', async t => {
```
Comment (sindresorhus) [inline test.js]:
```suggestion
test('non-human readable output - async', async t => {
```

File: index.d.ts [hunk 6]
```diff
@@ -1,7 +1,18 @@
+export type Options = {
+	/**
+	Modifies the output style.
+	If false returns value in recompilable source form.
+
```
Comment (sindresorhus) [inline index.d.ts:6]: Would be useful to link to the osascript docs about the different forms.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Don't forget to add it to the readme. Use this formatting: https://github.com/sindresorhus/boxen#options

File: test.js (hunk 5 above)
Comment (ruicalheno133) [reply test.js]: Fixed in [8ae5a3d](https://github.com/sindresorhus/run-applescript/pull/14/commits/8ae5a3dceab125fbda2be9ef95a515efbdfc40bc).

File: index.d.ts (hunk 6 above)
Comment (ruicalheno133) [reply index.d.ts:6]: Agree. I added the link to osascript's man page in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f)

File: index.js (hunk 4 above)
Comment (ruicalheno133) [reply index.js]: I feel like outputArgs is expressive enough but updated it in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f)

File: index.js (hunk 3 above)
Comment (ruicalheno133) [reply index.js]: I believe the original approach ends up being more readable and easy to extend.
Nevertheless, I updated it as suggested in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f), since I don't think there will be much more options to add.

File: index.d.ts (hunk 1 above)
Comment (ruicalheno133) [reply index.d.ts]: Should I keep the example in both the functions and `humanReadableOutput` docs?

Comment (sindresorhus) [reply index.d.ts]: No, only in `humanReadableOutput`.

Comment (ruicalheno133) [reply index.d.ts]: Done in cf16b68.

**Comment by sindresorhus** [comment]
> Don't forget to add it to the readme. Use this formatting: https://github.com/sindresorhus/boxen#options

File: readme.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ $ npm install run-applescript
 ## Usage
 
 ```js
-import {runAppleScript} from 'run-applescript';
+import { runAppleScript } from "run-applescript";
```
Comment (sindresorhus) [inline readme.md]: Don't make unrelated changes.

File: readme.md [hunk 8]
```diff
@@ -21,14 +21,48 @@ console.log(result);
...
+
+The script to run.
+
+#### options
+
+Type: `object`
+
+##### humanReadableOutput
+
+Type: `boolean`
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
Type: `boolean`\
```

File: readme.md [hunk 9]
```diff
@@ -21,14 +21,48 @@ console.log(result);
...
+Default: `true`
+
+Output modifier flag. When `false`, returns the value in a [recompilable source form](https://ss64.com/osx/osascript.html).
```
Comment (sindresorhus) [inline readme.md]: This is not the same as in index.d.ts

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: readme.md (hunk 7 above)
Comment (ruicalheno133) [reply readme.md]: Sure. Linter automatically "messed" this up. 
Fixed it.
